hax/hax/threadkv.py

```diff
@@ -0,0 +1,43 @@
+# Copyright (c) 2020 Seagate Technology LLC and/or its Affiliates
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+# For any questions about this software or licensing,
+# please email opensource@seagate.com or cortx-questions@seagate.com.
+#
+
+from threading import current_thread
+from typing import Any
+
+
+def get_thread_local_storage() -> dict:
+    return current_thread().__dict__
```

Comment:
Have you seen this document https://docs.python.org/3/library/threading.html#thread-local-data ?

Reply:
Yeah, I missed changing the earlier implementation. But overall I feel instead of caching kv in tls, may be its better to have a kvcache separately. That will help localising the caching mechanism and invalidating it would also be easier. Let me try this.

Reply:
@knekrasov , have updated the implementation. Kindly review.